app/code/Magento/Catalog/Block/Product/View/Gallery.php

```diff
@@ -28,19 +29,29 @@ class Gallery extends \Magento\Catalog\Block\Product\View\AbstractView
     protected $jsonEncoder;
 
     /**
-     * @param \Magento\Catalog\Block\Product\Context $context
-     * @param \Magento\Framework\Stdlib\ArrayUtils $arrayUtils
-     * @param EncoderInterface $jsonEncoder
-     * @param array $data
+     * @var DataObject
+     */
+    protected $galleryImagesConfig;
+
+    /**
+     * @param \Magento\Catalog\Block\Product\Context                              $context
+     * @param \Magento\Framework\Stdlib\ArrayUtils                                $arrayUtils
+     * @param \Magento\Framework\Json\EncoderInterface                            $jsonEncoder
+     * @param \Magento\Catalog\Model\Product\Gallery\ImagesConfigFactoryInterface $imagesConfigFactory
+     * @param array                                                               $galleryImagesConfig
+     * @param array                                                               $data
      */
     public function __construct(
         \Magento\Catalog\Block\Product\Context $context,
         \Magento\Framework\Stdlib\ArrayUtils $arrayUtils,
-        EncoderInterface $jsonEncoder,
+        \Magento\Framework\Json\EncoderInterface $jsonEncoder,
+        \Magento\Catalog\Model\Product\Gallery\ImagesConfigFactoryInterface $imagesConfigFactory,
```

Comment:
Unfortunately, due to backwards compatibility policy, we can not accept changes which change constructor signature by introducing a new required parameter.
You may fix it like:
```php
use Magento\Framework\App\ObjectManager;
...
public function __construct(
    \New\Dependency $newDependency = null
) {
    $this->newDependency = $newDependency ?: ObjectManager::getInstance()->get(\New\Dependency::class)
}
```